genesyscloud/external_contacts_organization/data_source_genesyscloud_externalcontacts_organization.go

```diff
@@ -0,0 +1,36 @@
+package external_contacts_organization
+
+import (
+	"context"
+	"fmt"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+func dataSourceExternalContactsOrganizationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := newExternalContactsOrganizationProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+
+	return util.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+		externalOrganizationId, retryable, response, err := proxy.getExternalContactsOrganizationIdByName(ctx, name)
+
+		if err != nil && !retryable {
+			return retry.NonRetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Error searching exteral organization %s | error: %s", name, err), response))
+		}
+
+		if retryable {
```

Comment:
Could you move this check inside the error check? That way we will only return an error if err is not nil. If for some reason `getExternalContactsOrganizationIdByName` returns nil for the error but true for retryable, then we will fail for no reason. That might not be possible based on how you wrote the code, but at least there's no risk of it happening in the future if we refactor here